File: Languages/passwordProtect.py
import os

def mkdir(name):
  os.system(f"mkdir {name}")
  
def cd(newdir):
  os.chdir(str(newdir))
  
initial_directory = os.getcwd()

def part_1():
  a = -1
  while a < 9:
    a = a + 1
    mkdir(a)

def part_2():
  a = -1
  while a < 9:
    a = a + 1
    cd(a)
    b = -1
    while b < 9:
      b = b + 1
      mkdir(b)
    cd("../")
    b = -1
  cd(initial_directory)

def part_3():
  a = -1
  while a < 9:
    a = a + 1
    b = -1
    while b < 9:
      b = b + 1
      cd(f"{a}/{b}")
      c = -1
      while c < 9:
        c = c + 1
        mkdir(c)
      cd(initial_directory)
      c = -1
  cd(initial_directory)
  
def part_4():
  a = -1
  while a < 9:
    a = a + 1
    b = -1
    while b < 9:
      b = b + 1
      c = -1
      while c < 9:
        c = c + 1
        cd(f"{a}/{b}/{c}")
        d = -1
        while d < 9:
          d = d + 1
          mkdir(d)
          os.system(f"touch {d}/.combo")
        cd(initial_directory)
      d = -1
    c = -1
  b = -1
cd(initial_directory)

def start():
  part_1()
  print("25% complete")
  part_2()
  print("50% complete")
  part_3()
  print("75% complete")
  part_4()
  print("100% complete")
  
start()

File: Languages/passwordProtect.js
const process = require('process');
const fs = require('fs')

function part1(){
  for (var a = 0; a <= 9; a++){
    fs.mkdirSync(String(a))
  }
}

function part2(){
  for (var a = 0; a <= 9; a++){
    for (var b = 0; b <= 9; b++){
      fs.mkdirSync(`${String(a)}/${String(b)}`)
    }
  }
}

function part3(){
  for (var a = 0; a <= 9; a++){
    for (var b = 0; b <= 9; b++){
      for (var c = 0; c <= 9; c++){
        fs.mkdirSync(`${String(a)}/${String(b)}/${String(c)}`)
      }
    }
  }
}

function part4(){
  for (var a = 0; a <= 9; a++){
    for (var b = 0; b <= 9; b++){
      for (var c = 0; c <= 9; c++){
        for (var d = 0; d <= 9; d++){
          fs.mkdirSync(`${String(a)}/${String(b)}/${String(c)}/${String(d)}`)
        }
      }
    }
  }
}

function start(){
  part1()
  console.log("25% complete")
  part2()
  console.log("50% complete")
  part3()
  console.log("75% complete")
  part4()
  console.log("100% complete")
}

start()